notebooks/how_to_pyscf.md

```diff
@@ -0,0 +1,117 @@
+
+You may want to register a determinant list after a CASCI/CASSCF, selected CI (sCI), or full CI (FCI) calculation. To avoid inconsistencies, it is recommended to save this data in a separate TREXIO file from the one used for RHF results.
+
+
+```python
+norb = 10
+nelec = 4
+                          
+# Initialize CASCI calculation with Selected CI as the solver
+mcscf = mcscf.casci_symm.CASCI(mf, norb, nelec)
+mcscf.fcisolver = selected_ci_spin0_symm.SelectedCI(rhf)
+
+#Perform  CASCI calculation
+mcscf.kernel()
+ 
+#Save determinants into TREXIO
+trexio_file = 'my_det.hdf5'
+det_to_trexio(mcscf, norb, nelec, trexio_file)
+```
+
+The ```det_to_trexio``` function registers the number of MO, number of alpha and beta electrions, total number of determinants, coefficients and the determinants in bitfield format.
+
+## Applying a Threshold for Determinants
+You can filter determinants by their coefficients, saving only those that exceed a specified threshold:
+
+```python
+# Define a coefficient threshold
+ci_threshold = 1e-1
+
+# Save filtered determinants to a TREXIO file
+trexio_file = 'filtered_determinants.hdf5'
+det_to_trexio(casci, norb, nelec, trexio_file, ci_threshold = ci_threshold )
+```
+
+This approach allows to filter determinants based on their importance, reducing the size of the output file and focusing on significant contributions.
+
+## Inspecting Determinants and Coefficients
+You can retrieve and display the determinants and coefficients directly using the following code:
+
+```python
+from itertools import islice
+
+# Retrieve determinants and coefficients
+occsa, occsb, ci_coeffs,num_determinants = get_occsa_and_occsb(casci, norb, nelec, ci_threshold = ci_threshold)
+
+# Print the first 10 determinants
+for a, b, coeff in islice(zip(occsa, occsb, ci_coeffs), 10):
+    print(f"   {a}          {b}         {coeff}")
+```
+If no ```ci_threshold``` is provided, the function processes the full list of determinants by default
+
+## Read determinant from TREXIO
+You may also want to import determinant information from a TREXIO file generated by other software. This can be done using the following code:
+
+```python
+num_det, coeff, det_list = read_det_trexio(trexio_file)
+```
+It retrives the number of determinantm coefficient and determinant list in bitfield format
+This data can then be analyzed or used in subsequent computations.
+
+## Import PySCF wavefunction to QP2
+QP2 (https://github.com/QuantumPackage/qp2/tree/dev-stable) requires additional information to import the PySCF - TREXIO file. The state group is necessary, including the total number of states (including the ground state) and the ID. For the ground state (GS), the ID is 0. Therefore, these lines should be added in your PySCF script if you plan to use its wavefunction into QP2:
+
+```python
+import trexio
+
+filename = 'data_for_qp2.hdf5'
+trexio_file = trexio.File(filename, mode='w', back_end=trexio.TREXIO_HDF5)
```

Comment:
You can use a context handling here, e.g. 

```python
with trexio.File(filename, mode='w', back_end=trexio.TREXIO_HDF5) as trexio_file:
  #Index of the current state: 0 is the GS
  trexio_file.set_state(0)
  #Total number of states
  trexio.write_state_num(trexio_file,1) 
```

This way, you do not need to manually close the file via `.close()`